EIPS/eip-5564.md

```diff
@@ -0,0 +1,110 @@
+Observers have no possibility of linking the recipient's stealth address to the recipient's identity, leaving only the sender with that information.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every smart contract compliant with this EIP wallet MUST implement the [EIP-165](./eip-165.md) (`0x01ffc9a7`) interface.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.6;
+...
+interface IERC5564 {
+    
+    /// @notice Public Key coordinates of the wallet owner
+    /// @dev Is used by other wallets to generate stealth addresses
+    ///  on behalf of the wallet owner.
+    bytes publicKey;
+
+
+    /// @notice Generates a stealth address that can be accessed only by the recipient.
+    /// @dev Function is executed locally by the sender on the recipient's wallet to
+    ///  generate a stealthAddress and publishableData S. The Caller/Sender must select a secret
+    ///  value s and compute the stealth address of the wallet owner and the matching public key S
+    ///  to the selected secret s.
+    /// @param secret A secret value selected by the sender
+    function generateStealthAddress(uint256 secret) returns (bytes publishableData, address stealthAddress)
+
+}
+
+interface PubStealthInfoContract {
+
+    /// @noticeImmutable contract that broadcasts an 
+    ///  event with the address of the stealthRecipient and 
+    ///  publishableData S for every privateTransfer. 
+    /// @dev Emits event with private transfer information S and the recipient's address.
+    ///  S is generated by the sender and represents the public key to the secret s.
+    ///  The sender broadcasts S for every private transfer. Users can use S to check if they were
+    ///  the recipients of a respective transfer by comparing it to stealthRecipient.
+    /// @param stealthRecipient The address to send the funds to
+    /// @param publishableData The public key to the sender's secret
+    event PrivateTransfer(address indexed stealthRecipient, bytes publishableData)
+}
+```
+
+#### Stealth Address Generation
+
+
+Sender Keypair    --> (s,S) | S = s * G 
+
+Recipient Keypair --> (p,P) | P = p * G
+
+NOTE: `s` represents a sender-generated secret
+
+
+#### For transfering:
+
+sharedSecret    = s*P
+
+stealthAddress  = pubtoaddr(P + (G * keccak(sharedSecret)))
+
+publishableData = G * s = S
+
+
+
+#### For receiving:
+
+for all PrivateTransfer Events do:
+
+if pubtoaddr(P + (G * keccak(S * p))) == stealthRecipient:
+
+store_key(p + keccak(S * p))
```

Comment:
I recommend you write this as e.g. a solidity function.

Reply:
Yeah, good idea! Makes it directly more clear.